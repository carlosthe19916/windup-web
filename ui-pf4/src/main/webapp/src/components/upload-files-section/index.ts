export { UploadFilesSection } from "./upload-files-section";
