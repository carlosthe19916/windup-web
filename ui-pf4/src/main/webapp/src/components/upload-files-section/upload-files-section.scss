.upload-files-section__component__dropzone {
  border: 2px solid var(--pf-global--Color--dark-200);
  border-style: dotted;
}
