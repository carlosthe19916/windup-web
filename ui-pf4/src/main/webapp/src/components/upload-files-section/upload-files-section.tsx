import * as React from "react";
import { useState } from "react";
import {
  EmptyState,
  EmptyStateBody,
  Button,
  EmptyStateVariant,
  Progress,
  ProgressSize,
  Stack,
  StackItem,
  ProgressVariant,
} from "@patternfly/react-core";
import { useDropzone } from "react-dropzone";
import { AxiosPromise, AxiosError } from "axios";

import "./upload-files-section.scss";

import { Application } from "../../models/api";
import { UploadFile } from "../upload-file";

export interface UploadFilesSectionProps {
  applications?: Application[];
  fileFormName: string;
  upload: (formData: FormData, config: any) => AxiosPromise;
  onSuccess?: (file: File) => void;
  onError?: (error: AxiosError, file: File) => void;
}

export const UploadFilesSection: React.FC<UploadFilesSectionProps> = ({
  applications,
  fileFormName,
  upload,
  onError,
  onSuccess,
}) => {
  const [files, setFiles] = useState<File[]>([]);

  const handleOnDrop = (acceptedFiles: File[]) => {
    // Add new files to the beginning of the array
    setFiles((current) => [...current, ...acceptedFiles]);
  };

  const { getRootProps, getInputProps, open } = useDropzone({
    noClick: true,
    noKeyboard: true,
    onDrop: handleOnDrop,
  });

  return (
    <React.Fragment>
      <Stack hasGutter>
        <StackItem>
          <EmptyState
            variant={EmptyStateVariant.small}
            {...getRootProps({
              className: "upload-files-section__component__dropzone dropzone",
            })}
          >
            <EmptyStateBody>
              Drag a file here or browse to upload
            </EmptyStateBody>
            <Button variant="primary" onClick={open}>
              Browser
            </Button>
            <input {...getInputProps()} />
          </EmptyState>
        </StackItem>
        <StackItem>
          <Stack hasGutter>
            {files.map((file: File, index) => (
              <StackItem key={index}>
                <UploadFile
                  file={file}
                  fileFormName={fileFormName}
                  startUpload={true}
                  upload={upload}
                  onSuccess={() => {
                    if (onSuccess) {
                      onSuccess(file);
                    }
                  }}
                  onError={(error) => {
                    if (onError) {
                      onError(error, file);
                    }
                  }}
                />
              </StackItem>
            ))}
            {(applications || []).map((a, index) => (
              <StackItem key={index}>
                <Progress
                  title={a.inputFilename}
                  size={ProgressSize.sm}
                  value={100}
                  variant={ProgressVariant.success}
                />
              </StackItem>
            ))}
          </Stack>
        </StackItem>
      </Stack>
    </React.Fragment>
  );
};
