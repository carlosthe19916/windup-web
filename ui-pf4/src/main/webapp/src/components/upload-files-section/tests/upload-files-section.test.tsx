import * as React from "react";
import { shallow } from "enzyme";
import { UploadFilesSection } from "../upload-files-section";

describe("Welcome", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<UploadFilesSection />);
    expect(wrapper).toMatchSnapshot();
  });
});
