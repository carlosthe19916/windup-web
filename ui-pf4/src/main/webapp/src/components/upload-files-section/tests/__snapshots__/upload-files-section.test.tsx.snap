// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Renders without crashing 1`] = `
<Fragment>
  <Stack
    hasGutter={true}
  >
    <StackItem>
      <EmptyState
        className="upload-files-section__component__dropzone dropzone"
        onBlur={null}
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        onFocus={null}
        onKeyDown={null}
        variant="small"
      >
        <EmptyStateBody>
          Drag a file here or browse to upload
        </EmptyStateBody>
        <Button
          onClick={[Function]}
          variant="primary"
        >
          Browser
        </Button>
        <input
          autoComplete="off"
          multiple={true}
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          tabIndex={-1}
          type="file"
        />
      </EmptyState>
    </StackItem>
    <StackItem>
      <Stack
        hasGutter={true}
      />
    </StackItem>
  </Stack>
</Fragment>
`;
